bundles/io/org.eclipse.smarthome.io.transport.mqtt.test/src/test/java/org/eclipse/smarthome/io/transport/mqtt/test/MqttBrokerConnectionTests.java

```diff
@@ -28,125 +29,77 @@
-            MqttBrokerConnection b = new MqttBrokerConnection("name", "ssl://123.123.123.123", true);
-            assertFalse(a.isTextualConfiguredBroker());
-            assertTrue(b.isTextualConfiguredBroker());
-        } catch (ConfigurationException e) {
-            fail("Unexpected configuration exception: " + e.getMessage());
-        }
-        // Test an invalid URL
-        try {
-            @SuppressWarnings("unused")
-            MqttBrokerConnection ignored = new MqttBrokerConnection("name", "unsupported://123.123.123.123", false);
-            fail("Configuration exception expected: No valid url protocol");
-        } catch (ConfigurationException e) {
-        }
-        // Test an invalid name
-        try {
-            @SuppressWarnings("unused")
-            MqttBrokerConnection ignored = new MqttBrokerConnection(" ", "tcp://123.123.123.123", false);
-            fail("Configuration exception expected: No valid name");
-        } catch (ConfigurationException e) {
-        }
+        MqttBrokerConnection a = new MqttBrokerConnection("name", "tcp://123.123.123.123", false);
+        MqttBrokerConnection b = new MqttBrokerConnection("name", "ssl://123.123.123.123", true);
+        assertFalse(a.isTextualConfiguredBroker());
+        assertTrue(b.isTextualConfiguredBroker());
+    }
+
+    @Test(expected = ConfigurationException.class)
+    public void testConstructorInvalidProtocol() throws ConfigurationException {
+        new MqttBrokerConnection("name", "unsupported://123.123.123.123", false);
+    }
+
+    @Test(expected = ConfigurationException.class)
+    public void testConstructorInvalidName() throws ConfigurationException, MqttException {
+        new MqttBrokerConnection(" ", "tcp://123.123.123.123", false);
     }
 
     @Test
-    public void messageConsumerTests() {
+    public void messageConsumerTests() throws ConfigurationException, MqttException {
         final String url = "tcp://123.123.123.123";
         final String name = "TestName12@!";
-        MqttBrokerConnection a;
-        try {
-            a = new MqttBrokerConnection(name, url, false);
-        } catch (ConfigurationException e) {
-            fail();
-            return;
-        }
+        MqttBrokerConnection a = new MqttBrokerConnection(name, url, false);
         assertFalse(a.isConsumers());
-        MqttMessageSubscriber subscriber = new MqttMessageSubscriber() {
-            @Override
-            public void processMessage(String topic, byte[] payload) {
-
-            }
-
-            @Override
-            public String getTopic() {
-                return "topic";
-
-            }
-        };
-        try {
-            a.addConsumer(subscriber);
-        } catch (MqttException e) {
-            fail("addConsumer failed");
-        }
+        MqttMessageSubscriber subscriber = mock(MqttMessageSubscriber.class);
+        when(subscriber.getTopic()).thenAnswer(i -> "topic");
```

Comment:
Nice! alternative (no need for you to change it here):

`when(subscriber.getTopic()).thenReturn("topic");`